### ISSUE TITLE
AdjacentVertexFilterOptimizerStrategy can stuck infinitely

### ISSUE THREAD

**Issue by mad** [JanusGraph/janusgraph#1500]
Sample of code
```
graph = JanusGraphFactory.open("inmemory")
vertex = graph.traversal().addV().next()
graph.traversal().V().outE().has("p", "v").filter(inV().is(vertex)).tryNext()
```
Last line infinitely stuck

**Comment by FlorianHockmann** [comment]
`tryNext()` doesn't return here or what is the behaviour you're seeing?

**Comment by porunov** [comment]
Fixed by #1501
